### PR TITLE
Fix handling of URL in file field

### PR DESCRIPTION
**Pull request by koppor**
~~One test introduced in #7043 raises an exception on Windows, because `:` is not allowed as path separator.~~

~~This PR adds a workaround.~~

~~Alternative: `@DisableOnWindows` test annotation. Since we (hopefully) need that only once, we decided for a quick workaround by checking for `OS.WINDOWS`.~~

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by Siedlerchr — DISMISSED** [review]

**Review by calixtus — APPROVED** [review]

File: src/test/java/org/jabref/logic/bibtex/FileFieldWriterTest.java [hunk 1]
```diff
@@ -47,8 +50,16 @@ public void parseCorrectOnlineInput() throws MalformedURLException {
...
 
+        if (OS.WINDOWS) {
+            // on Windows, a ":" in a path cannot be used
+            // java.nio.file.InvalidPathException: Illegal char <:> at index 3: htt://arxiv.org/pdf/2010.08497v1
+            // Thus, we assert that statement and return
+            assertThrows(InvalidPathException.class, () -> Path.of(inputURL));
+            return;
+        }
+
+        List<LinkedFile> expected = Collections.singletonList(new LinkedFile("", Path.of(inputURL), "PDF"));
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/bibtex/FileFieldWriterTest.java:62]: What is `FileFieldParser.parse(input)` returning on Windows?

Comment (koppor) [reply src/test/java/org/jabref/logic/bibtex/FileFieldWriterTest.java:62]: It also returns an exception. Changed the `assertThrows` accordingly.

The issue is at `field = new LinkedFile(entry.get(0), Path.of(entry.get(1)), entry.get(2));`, where `Path.of(...)` is called. Future work? 😇

Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/bibtex/FileFieldWriterTest.java:62]: We have something like LinkedFile.isOnlineLink as check in some places

Comment (tobiasdiez) [reply src/test/java/org/jabref/logic/bibtex/FileFieldWriterTest.java:62]: I think you found a bug then. The string should always be parsed to a LinkedFile, even if it's not a valid path or url. The fix should be simple as well: just introduce a constructor accepting the path also a string, and change the filefieldparser to catch the invalidpathexception as well.

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

**Comment by koppor** [comment]
Bug should be resolved. Changes added.

Fixes https://github.com/JabRef/jabref/issues/7359. Added test cases for that, too.

**Review by Siedlerchr — APPROVED** [review]

**Review by calixtus — DISMISSED** [review]
Seems to me the requested changes have been addressed and the code looks good to me. So...

**Comment by calixtus** [comment]
Well, a test is still failing.

**Comment by koppor** [comment]
Had to fix a checkstyle issue.

Also added a heuristics for `//` so that the thing is consistent on Linux and Windows. --> `//` is kept when converting to an internal representation (before, on windows, it was kept, on Linux, it was shrunk to `/`).

Should be ready now.

**Review by calixtus — APPROVED** [review]

**Review by tobiasdiez — APPROVED** [review]
LGTM

Maybe it's time to introduce OnlineLinkedFile, PathLinkedFile and InvalidLinkedFile deriving from an abstract LinkedFile class (as we currently parse a string to a LinkedFile trying to convert strings to paths and urls, but then LinkedFile converts them to a string again...). Something for another time.

**Comment by calixtus** [comment]
For the record: Fixes #6859

**Pull request merged**